### ISSUE TITLE
Remodeling our data model

### ISSUE THREAD

**Issue by ciearius** [GeneralMine/S2QUAT#24]
As stated in #23 I've gone ahead and added a first prototype of the basic model (546b493aa4a3aeeee0ec25e9d2394e798aaf20e3). I did not pull it out of the database on purpose to not generate bloat.

> When working on this new take, please bear in mind to use a non-generalized way! We do not want ONE table/model for all, as this is what we want to avoid/migrate away from!

I've added some `TODOs`. Some of them boil down to the following questions/uncertainties:
- clarify what `roles` do and how they are used.
- how do we want to implement the new functionality?

**Comment by ciearius** [comment]
I need you on this with me @GeneralMine!

**Comment by GeneralMine** [comment]
Really good! Thanks for this setup! 🙏🙏🙏
I will update the database model asap so that you can continue working on prisma.

To answer your questions:
- Roles
    - User: A normal user with access to all the interview and questions functionality
    - Editor: User with the ability to edit the qualityModel table.
    - Admin: Editor with the ability to administrate the qualityModel and users table
- I'm working on the new database model, so I will inform you when the diagram updated!

**Comment by ciearius** [comment]
So basically: `roles` is an enum? Does it need to be specific to the project? Or are we ok with it being across all his projects?

**Comment by GeneralMine** [comment]
The new database model so far:
![Datenbankmodell](https://user-images.githubusercontent.com/33132009/124042577-6b9dda00-da09-11eb-9d9c-ca6a525065e8.png)

I switched from ER Diagram to a class diagram, due to easier and faster drawing capabilities. 
- unidirectional associations: `1:n relationship, directing to the entity of which the key is stored`
- enums: `a attribute that can only have enum values`
- aggregation: `1:n relationship` where the key lays on the other side. This is more of a semantic difference rather than a technical one. Thats meant like "This entity consists of n others"
- switched to german for better semantic understanding with the words

Big changes:
- I removed the employees, because its not really useful right now
- seperated qualitymodel from the questions. First is then copied into a question when used
- splitted questions into the 4 (Feld, Merkmal, Faktor, Kriterium) or 3 layers (Seiten, Kategorie, Frage)
- Fragebogen is a additional layer of abstraction to allow multiple Fragebögen per Szenario
